pkg/simulation/simulation.go

```diff
@@ -59,22 +59,22 @@ func (sE *SimEnv) Simulate() {
 			agentIndex++
 		}
 	}
-	a.LimitIterations(sE.Iterations)
+	a.LimitIterations(sE.dayInfo.TotalIterations)
 	sE.Log("Simulation Started")
 	sE.simulationLoop(a, t)
 	sE.Log("Simulation Ended")
 }
 
 func (sE *SimEnv) simulationLoop(a *abm.ABM, t *infra.Tower) {
-	for i := 1; i <= sE.Iterations; i++ {
+	for sE.dayInfo.CurrTick <= sE.dayInfo.TotalIterations {
 		missingAgentsMap := t.UpdateMissingAgents()
 		for floor := range missingAgentsMap {
 			for _, agentType := range missingAgentsMap[floor] {
 				sE.createNewAgent(a, t, agentType, floor)
 			}
 		}
-
-		a.SimulationIterate(i)
+		sE.dayInfo.CurrTick++
```

Comment:
Why not just put this into the for loop?

Reply:
Not currently integrated into ABM, and pretty much the same as putting it into a Tower tick, I guess this is in the overall for loop.

Reply:
Never mind, got confused.

I think it's a bit weird to not have the tick increment as the last statement in the loop. If we initialise CurrTick to 1 then move the increment to the end of the loop, would that still work?

Reply:
Sounds good👍